url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+	public HiveLexer(CharStream input) {
+		this(input, new RecognizerSharedState());
+	}
+	public HiveLexer(CharStream input, RecognizerSharedState state) {
+		super(input,state);
+	}
+	@Override public String getGrammarFileName() { return "HiveLexer.g"; }
+
+	// $ANTLR start "KW_TRUE"
+	public final void mKW_TRUE() throws RecognitionException {
+		try {
+			int _type = KW_TRUE;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:23:9: ( 'TRUE' )
+			// HiveLexer.g:23:11: 'TRUE'
+			{
+			match("TRUE"); 
+
+			}
+
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "KW_TRUE"
+
+	// $ANTLR start "KW_FALSE"
+	public final void mKW_FALSE() throws RecognitionException {
+		try {
+			int _type = KW_FALSE;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:24:10: ( 'FALSE' )
+			// HiveLexer.g:24:12: 'FALSE'
+			{
+			match("FALSE"); 
+
+			}
+
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "KW_FALSE"
+
+	// $ANTLR start "KW_ALL"
+	public final void mKW_ALL() throws RecognitionException {
+		try {
+			int _type = KW_ALL;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:25:8: ( 'ALL' )
+			// HiveLexer.g:25:10: 'ALL'
+			{
+			match("ALL"); 
+
+			}
+
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "KW_ALL"
+
+	// $ANTLR start "KW_AND"
+	public final void mKW_AND() throws RecognitionException {
+		try {
+			int _type = KW_AND;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:26:8: ( 'AND' )
+			// HiveLexer.g:26:10: 'AND'
+			{
+			match("AND"); 
+
+			}
+
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "KW_AND"
+
+	// $ANTLR start "KW_OR"
+	public final void mKW_OR() throws RecognitionException {
+		try {
+			int _type = KW_OR;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:27:7: ( 'OR' )
+			// HiveLexer.g:27:9: 'OR'
+			{
+			match("OR"); 
+
+			}
+
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "KW_OR"
+
+	// $ANTLR start "KW_NOT"
+	public final void mKW_NOT() throws RecognitionException {
+		try {
+			int _type = KW_NOT;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:28:8: ( 'NOT' | '!' )
+			int alt1=2;
+			int LA1_0 = input.LA(1);
+			if ( (LA1_0=='N') ) {
+				alt1=1;
+			}
+			else if ( (LA1_0=='!') ) {
+				alt1=2;
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 1, 0, input);
+				throw nvae;
+			}
+
+			switch (alt1) {
+				case 1 :
+					// HiveLexer.g:28:10: 'NOT'
+					{
+					match("NOT"); 
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:28:18: '!'
+					{
+					match('!'); 
+					}
+					break;
+
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 493:508, 5801:5816, 5868:5884, 9307:9323, 9566:9582.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.